src/reports/data/node.hpp

```diff
@@ -1,28 +1,22 @@
 #pragma once
 
+#include <cstdint>
 #include <vector>
 
 typedef double* (*refresh_function_t)(double*);
 
 class Node {
-  private:
-    uint64_t m_gid;
-    uint64_t m_vgid;
-    uint32_t m_element_id;
-
-    std::vector<uint32_t > m_element_ids;
-    std::vector<double*> m_elements;
-
 public:
-    Node(uint64_t gid, uint64_t vgid);
```

Comment:
I understand you removed the constructor because this is not used anywhere.
However I think it makes sense to keep it and use it in the sonata_data.cpp instead of the key of the map.
`
void prepare_dataset() {
    for(auto& kv: *m_nodes) {
     ....
        // m_node_ids.push_back(kv.first);
        m_node_ids.push_back(kv.second.get_gid());
    }
}
`

Reply:
`gid` and `vgid` are not used anywhere, so why keep them?
I remove the constructor because the default one is enough.

Reply:
But as I said, right now the key of the map of nodes is being used to identify the gid, however i think it makes more sense to have this information in the node itself.